dimod/core/sampler.py

```diff
@@ -238,3 +239,22 @@ def sample_qubo(self, Q, **parameters):
         """
         bqm = BinaryQuadraticModel.from_qubo(Q)
         return self.sample(bqm, **parameters)
+
+    def remove_unknown_kwargs(self, **kwargs):
+        """Check that all `kwargs` are accepted by the sampler. If a
+        keyword is unknown, a warning is raised and the argument is removed.
+
+        Args:
+            **kwargs:
+                Keyword arguments to be validated.
+
+        Returns:
+            dict: Updated `kwargs`
+
+        """
+        for kw in [k for k in kwargs if k not in self.parameters]:
+            msg = "Ignoring unknown kwarg: {!r}".format(kw)
+            warnings.warn(msg, SamplerUnknownArgWarning, stacklevel=3)
+            kwargs.pop(kw)
+
+        return kwargs
```

Comment:
I find this confusing. `kwargs` are modified inplace, and that modified/original dict is returned.

Can we either:
1) return a shallow copy, with unknown items removed, or
2) not return anything (like `dict.update`)

Reply:
Please ignore my previous comment. `kwargs` are constructed *on call*, even if expanded from `**kwargs`, so the change to them can't be propagated to the caller.